### ISSUE TITLE
Catch UnicodeDecodeError during authentication to handle non-ASCII hostnames on Windows

### ISSUE THREAD

**Issue by brianhou** [Cal-CS-61A-Staff/ok-client#181]
We have a current student who ran into some issues with the client while having a Korean host name (on Windows, if that's significant). Changing the host name to something in English seemed to fix the issue, although I'm not entirely sure what the issue is. That's not a great solution, so it would be great if we could figure out how to fix ok for future semesters.

```
$ python ok --debug --authenticate
DEBUG | ok.py:103 | Namespace(all=False, authenticate=True, backup=F alse, config=None, debug=True, export=False, insecure=False, interact ive=False, latest=False, local=False, lock=False, no_hints=False, no_ update=False, question=None, restore=False, score=False, score_out=<c lient.utils.output._OutputLogger object at 0x000001CFBC8807F0>, serve r='ok.cs61a.org', submit=False, tests=False, timeout=10, unlock=False , update=False, verbose=False, version=False)
DEBUG | ok.py:107 | Python 64bit
INFO | auth.py:120 | SSL bindings are available.
Please enter your bCourses email.
bCourses email: ***
INFO | software_update.py:20 | Existing OK version: v1.6.2
INFO | software_update.py:21 | Checking latest version from https://ok.cs61a.org/api/v3/version/ok-client
Traceback (most recent call last):
File "C:\Users\***\AppData\Local\Programs\Python\Python35\lib\runpy.py", line 170, in _run_module_as_main
"__main__", mod_spec)
File "C:\Users\***\AppData\Local\Programs\Python\Python35\lib\runpy.py", line 85, in _run_code
exec(code, run_globals)
File "ok\__main__.py", line 9, in <module>
File "ok\client\cli\ok.py", line 121, in main
File "ok\client\utils\auth.py", line 218, in authenticate
File "C:\Users\***\AppData\Local\Programs\Python\Python35\lib\socketserver.py", line 443, in __init__
self.server_bind()
File "C:\Users\***\AppData\Local\Programs\Python\Python35\lib\http\server.py", line 140, in server_bind
self.server_name = socket.getfqdn(host)
File "C:\Users\***\AppData\Local\Programs\Python\Python35\lib\socket.py", line 662, in getfqdn
hostname, aliases, ipaddrs = gethostbyaddr(name)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc0 in position 0: invalid start byte
Checking for software updates...
OK is up to date
```

**Comment by Sumukh** [comment]
Seems like an python bug that was resolved in 3.5.2 - (which was released yesterday). 

Ok should catch this exception and tell students how to fix it (either upgrade to 3.5.2 or change the hostname) 

https://hg.python.org/cpython/rev/0681f0a1fe6e
http://bugs.python.org/issue26227
https://docs.python.org/3.5/whatsnew/changelog.html#python-3-5-2

**Comment by soumyabasu** [comment]
Yeah- I looked into this issue a few years ago and it was a python bug.

Glad to hear they fixed it

On Tuesday, June 28, 2016, Sumukh Sridhara <notifications@github.com> wrote:

> Seems like an python bug that was resolved in 3.5.2 - (which was released
> yesterday).
>
> Ok should catch this exception and tell students how to fix it (either
> upgrade to 3.5.2 or change the hostname)
>
> https://hg.python.org/cpython/rev/0681f0a1fe6e
> http://bugs.python.org/issue26227
> https://docs.python.org/3.5/whatsnew/changelog.html#python-3-5-2
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Cal-CS-61A-Staff/ok-client/issues/181#issuecomment-229224483>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABTAfoUV_LrEP6MRLA-BDL2Ih2Q86xfKks5qQb2SgaJpZM4JAP5p>
> .
>


-- 
Sent from Phone

**Comment by brianhou** [comment]
Huh, what do ya know. That's cool. I'll let that student know.

**Comment by knrafto** [comment]
Did upgrading to 3.5.2 work for the student? If so, I'd like to close this issue.

**Comment by brianhou** [comment]
I never heard back from them about it; I think they just permanently changed their hostname to English.

**Comment by epai** [comment]
Closed by #312
